test/jdk/java/util/prefs/AddNodeChangeListener.java

```diff
@@ -34,75 +34,88 @@
 public class AddNodeChangeListener {
 
     private static final int SLEEP_ITRS = 10;
+    private static final String N2_STR = "N2";
     private static boolean failed = false;
     private static Preferences userRoot, N2;
     private static NodeChangeListenerAdd ncla;
 
     public static void main(String[] args)
              throws BackingStoreException, InterruptedException {
-        userRoot = Preferences.userRoot();
-        ncla = new NodeChangeListenerAdd();
-        userRoot.addNodeChangeListener(ncla);
-        //Should initiate a node added event
-        addNode();
-        // Should not initiate a node added event
-        addNode();
-        //Should initate a child removed event
-        removeNode();
-
-        if (failed) {
-            throw new RuntimeException("Failed");
+        try {
+            userRoot = Preferences.userRoot();
+            // Make sure test node is not present before test
+            clearPrefs();
+
+            ncla = new NodeChangeListenerAdd();
+            userRoot.addNodeChangeListener(ncla);
+            //Should initiate a node added event
+            addNode();
+            // Should not initiate a node added event
+            addNode();
+            //Should initate a child removed event
+            removeNode();
+
+            if (failed) {
+                throw new RuntimeException("Failed");
+            }
+        } finally {
+            // Make sure test node is not present after the test
+            clearPrefs();
```

Comment:
If this call throws a `RuntimeException("Unable to clear...")`, it will override the possible `RuntimeException("Failed") in the `try` block, but I think it is OK.

Reply:
That's true - I hadn't considered that. I think it's OK, too. Even so, I've added a message to the exception thrown from clearPrefs() to indicate if the test failed.